metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/execution/WorkflowExecutor.java

```diff
@@ -237,7 +243,30 @@ private void periodicCheckingLoop(long sleepTime, AbstractMetisPlugin abstractMe
     preparePluginStateAndFinishedDate(abstractMetisPlugin, taskState,
         consecutiveCancelOrMonitorFailures);
   }
-  
+
+  private boolean isMinuteCapOverWithoutChangeInProcessedRecords(
+      AbstractMetisPlugin abstractMetisPlugin, AtomicInteger previousProcessedRecords,
+      AtomicLong stableRecordCountPeriodCounterInSeconds) {
+    final int processedRecords = abstractMetisPlugin.getExecutionProgress().getProcessedRecords();
+    //If CLEANING is in progress then just reset the values to be sure and return false
+    //Or if we have progress
+    if (abstractMetisPlugin.getPluginStatus() == PluginStatus.CLEANING
+        || previousProcessedRecords.get() != processedRecords) {
+      stableRecordCountPeriodCounterInSeconds.set(0L);
+      previousProcessedRecords.set(processedRecords);
+      return false;
+    }
+
+    final boolean isMinuteCapOverWithoutChangeInProcessedRecords =
+        stableRecordCountPeriodCounterInSeconds.addAndGet(monitorCheckIntervalInSecs)
```

Comment:
This doesn't work properly. You add the monitor check interval in the assumption that this is the amount of time that passed since the last time you came here. But the amount of time that passed is probably significantly bigger. 

I think you should maintain the reset time instead (i.e. in line 255 set System.currentTimeMillis(), so that here you only need to check how much time passed since then).